Adds support for construction of a TopicModel instance with zero numThreads

sagemintblue opened the pull request:
... in case client doesn't want internal thread pool initialized

Future work should improve client management of topic model thread pool e.g. to be able to shut it down.

sagemintblue committed:
Adds support for construction of a TopicModel instance with zero numThreads, in case client doesn't want internal thread pool initialized

Future work should improve client management of topic model thread pool e.g. to be able to shut it down.

jakemannix commented on core/src/main/java/org/apache/mahout/clustering/lda/cvb/TopicModel.java:
```diff
@@ -153,6 +153,12 @@ private static Vector viewRowSums(Matrix m) {
   }
 
   private void initializeThreadPool() {
+    // avoid initializing thread pool if client has specified less than one thread
```
I think this will mean that methods like update() will fail by silently blocking forever.  Not sure if that's a good idea.

sagemintblue replied:
Oof. Right. I've added a Precondition there.

sagemintblue committed:
Adds Precondition to TopicModel.update(Matrix)

This will catch poorly behaved clients who request less than one thread on TopicModel instantiation, but then attempt to use the multithreaded update method.

sagemintblue commented:
Any further thoughts on this branch?
